src/Microsoft.ML.Data/Training/TrainerInputBase.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Data.DataView;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.Data.IO;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Internal.Calibration;
+
+namespace Microsoft.ML.Trainers
+{
+    /// <summary>
+    /// The base class for all learner inputs.
+    /// </summary>
+    [TlcModule.EntryPointKind(typeof(CommonInputs.ITrainerInput))]
+    public abstract class TrainerInputBase
+    {
+        /// <summary>
+        /// The data to be used for training. Used only in entry-points, since in the API the expected mechanism is
+        /// that the user will use the <see cref="IEstimator{TTransformer}.Fit(IDataView)"/> or some other train
+        /// method.
+        /// </summary>
+        [BestFriend]
+        [Argument(ArgumentType.Required, ShortName = "data", HelpText = "The data to be used for training", SortOrder = 1, Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        internal IDataView TrainingData;
+
+        /// <summary>
+        /// Column to use for features.
+        /// </summary>
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Column to use for features", ShortName = "feat", SortOrder = 2, Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+        public string FeatureColumn = DefaultColumnNames.Features;
```

Comment:
>FeatureColumn [](start = 22, length = 13)

[https://github.com/dotnet/machinelearning/issues/2760](https://github.com/dotnet/machinelearning/issues/2760) if you want to can take care of this one as well.
Or someone else can take care of that later.

Reply:
ok

---
In reply to: [260868710](https://github.com/dotnet/machinelearning/pull/2748#discussion_r260868710) [](ancestors = 260868710)